python/pyspark/sql/tests/test_types.py

```diff
@@ -739,6 +739,11 @@ def test_timestamp_microsecond(self):
         tst = TimestampType()
         self.assertEqual(tst.toInternal(datetime.datetime.max) % 1000000, 999999)
 
+    # regression test for SPARK-23299
+    def test_row_without_column_name(self):
+        row = Row("Alice", 11)
+        self.assertEqual(repr(row), "<Row(Alice, 11)>")
```

Comment:
https://github.com/apache/spark/pull/20503#discussion_r225846381

Reply:
>  test non-ascii compatible characters

I have added a test with unicode values.  Of course that breaks in Python 2.7 because %s is used in __repr__.  I think it would be reasonable to change the use of %s to %r for representing the individual tuple values, just as was suggested originally in the JIRA ticket.  I've made that change and adapted the doctest accordingly.  What do you think about this?